cloud/amazon/ec2_vpc_peer.py

```diff
@@ -0,0 +1,1544 @@
+            }
+        ]
+        if status_codes:
+            params['Filters'].append(
+                {
+                    'Name': 'status-code',
+                    'Values': status_codes
+                }
+            )
+
+    elif vpc_peering_id:
+        params['VpcPeeringConnectionIds'] = [vpc_peering_id]
+        if status_codes:
+            params['Filters'] = [
+                {
+                    'Name': 'status-code',
+                    'Values': status_codes
+                }
+            ]
+
+    try:
+        result = (
+            client.describe_vpc_peering_connections(**params)
+            ['VpcPeeringConnections']
+        )
+        success = True
+
+    except botocore.exceptions.ClientError, e:
+        if e.response['Error']['Code'] == 'DryRunOperation':
+            success = True
+            err_msg = e.message
+        else:
+            err_msg = str(e)
+
+    return success, err_msg, result
+
+def is_active(peering_conn):
+    return peering_conn['status']['code'] == 'active'
+
+def is_deleted(peering_conn):
+    return peering_conn['status']['code'] == 'deleted'
+
+def is_expired(peering_conn):
+    return peering_conn['status']['code'] == 'expired'
+
+def is_failed(peering_conn):
+    return peering_conn['status']['code'] == 'failed'
+
+def is_initiating_request(peering_conn):
+    return peering_conn['status']['code'] == 'initiating-request'
+
+def is_pending(peering_conn):
+    return peering_conn['status']['code'] == 'pending-acceptance'
+
+def is_provisioning(peering_conn):
+    return peering_conn['status']['code'] == 'provisioning'
+
+def is_rejected(peering_conn):
+    return peering_conn['status']['code'] == 'rejected'
+
+def make_tags_in_proper_format(tags):
+    """Take a list of aws tags and convert them into a list of dictionaries.
+       Where the key is the actual key and not Key.
+    Args:
+        tags (list): The tags you want applied.
+
+    Basic Usage:
+        >>> tags = [{u'Key': 'env', u'Value': 'development'}]
+        >>> make_tags_in_proper_format(tags)
+        [
+            {
+               "env": "development",
+            }
+        ]
+
+    Returns:
+        List
+    """
+    formatted_tags = list()
+    for tag in tags:
+        formatted_tags.append(
+            {
+                tag.get('Key'): tag.get('Value')
+            }
+        )
+
+    return formatted_tags
+
+def convert_list_of_tags(tags):
+    """Convert a list of AWS Tag dictionaries into a dictionary.
+    Args:
+        tags (list): The tags you want applied.
+
+    Basic Usage:
+        >>> tags = [{u'Key': 'env', u'Value': 'development'}]
+        >>> convert_list_of_tags(tags)
+        {
+            "env": "development",
+        }
+
+    Returns:
+        Dict
+    """
+    converted_tags = dict()
+    for tag in tags:
+        tag = convert_to_lower(tag)
+        converted_tags[tag.get('key')] = tag.get('value')
+
+    return converted_tags
+
+def make_tags_in_aws_format(tags):
```

Comment:
Can we use helper function here too? https://github.com/ansible/ansible-modules-extras/blob/devel/cloud/amazon/GUIDELINES.md#ansible_dict_to_boto3_tag_list